ingress/controllers/nginx/configuration.md

```diff
@@ -198,6 +198,9 @@ http://nginx.org/en/docs/http/ngx_http_core_module.html#keepalive_timeout
 **proxy-send-timeout:** Sets the timeout in seconds for [transmitting a request to the proxied server](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_send_timeout). The timeout is set only between two successive write operations, not for the transmission of the whole request.
 
 
+**proxy-buffer-size:** Sets the size of the buffer used for [reading the first part of the response](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_buffer_size) received from the proxied server. This part usually contains a small response header.
```

Comment:
I picked 4k as the smaller of the two defaults. To be honest I don't have enough operational experience with nginx to give meaningful advice on the tradeoffs involved. This PR was motivated by the need to configure `proxy_buffer_size` to fix the `upstream sent too big header while reading response header from upstream` errors that we encountered in our deployment.

I'm happy to update this PR to provide the ability to also configure `proxy_buffering` and  `proxy_buffers` in addition to `proxy_buffer_size`. Should I go ahead and do that?

`proxy_buffering` is currently `off` as configured by the nginx ingress controller.